integration-tests/features/steps/stack_analyses_v2.py

```diff
@@ -234,6 +271,157 @@ def verify_vulnerability_count(context, component, vulnerability, expected):
     assert is_component_present, 'Component not present'
 
 
+def verify_dependency_count(context, component, expected):
+    """Check dependency count for given component."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'analyzed_dependencies')
+
+    is_component_present = False
+    for dep in json_data['analyzed_dependencies']:
+        if dep['name'] == component:
+            is_component_present = True
+            dependencies = dep['dependencies']
+            print(dep)
+            actual = len(dependencies)
+            assert actual == expected, \
+                'Component {} have {} deps but expected {}'.format(component, actual,
+                                                                   expected)
+    assert is_component_present, 'Component not present'
+
+
+def verify_valid_data_companion(context, component, latest_version):
+    """Check valid data in companion packages."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'recommendation')
+
+    is_component_present = False
+    for dep in json_data['recommendation']['companion']:
+        if dep['name'] == component:
+            is_component_present = True
+            actual_version = dep['latest_version']
+            assert actual_version == latest_version, \
+                'Component {} have {} but expected {}'.format(component, actual_version,
+                                                              latest_version)
+    assert is_component_present, 'Component not present'
+
+
+def check_essentials(vuln):
```

Comment:
There is already one function check_vulenrability_attributes() in this file, which does this, can we re-use it / extended it if required.

Reply:
okay i clearly missed this one